Sources/TuistEnvKit/Installer/EnvInstaller.swift

```diff
@@ -0,0 +1,108 @@
+import Foundation
+import TSCBasic
+import TuistSupport
+
+/// Protocol that defines the interface of an instance that can install versions of TuistEnv.
+protocol EnvInstalling: AnyObject {
+    /// It installs a version of Tuist in the local environment.
+    ///
+    /// - Parameters:
+    ///   - version: Version to be installed.
+    /// - Throws: An error if the installation fails.
+    func install(version: String) throws
+}
+
+/// Error thrown by the installer.
+///
+/// - versionNotFound: When the specified version cannot be found.
+/// - incompatibleSwiftVersion: When the environment Swift version is incompatible with the Swift version Tuist has been compiled with.
+enum EnvInstallerError: FatalError, Equatable {
+    case versionNotFound(String)
+    case incompatibleSwiftVersion(local: String, expected: String)
+
+    var type: ErrorType {
+        switch self {
+        case .versionNotFound: return .abort
+        case .incompatibleSwiftVersion: return .abort
+        }
+    }
+
+    var description: String {
+        switch self {
+        case let .versionNotFound(version):
+            return "Version \(version) not found"
+        case let .incompatibleSwiftVersion(local, expected):
+            return "Found \(local) Swift version but expected \(expected)"
+        }
+    }
+
+    static func == (lhs: EnvInstallerError, rhs: EnvInstallerError) -> Bool {
+        switch (lhs, rhs) {
+        case let (.versionNotFound(lhsVersion), .versionNotFound(rhsVersion)):
+            return lhsVersion == rhsVersion
+        case let (.incompatibleSwiftVersion(lhsLocal, lhsExpected), .incompatibleSwiftVersion(rhsLocal, rhsExpected)):
+            return lhsLocal == rhsLocal && lhsExpected == rhsExpected
+        default:
+            return false
+        }
+    }
```

Comment:
Isn't this synthesized by the compiler?

Reply:
It should be. I took this from `Installer.swift`, so I'll make the changes there as well.